generate_id.go

```diff
@@ -0,0 +1,86 @@
+package ravendb
+
+import (
+	"reflect"
+	"strings"
+)
+
+func tryGetIdFromInstance(entity interface{}) (string, bool) {
+	return tryGetIDFromInstance(entity)
+}
+
+// tryGetIDFromInstance returns value of ID field on struct if it's of type
+// string. Returns empty string if there's no ID field or it's not string
+func tryGetIDFromInstance(entity interface{}) (string, bool) {
+	rv := reflect.ValueOf(entity)
+	for rv.Kind() == reflect.Ptr {
+		rv = rv.Elem()
+	}
+	if rv.Kind() != reflect.Struct {
+		// TODO: maybe panic?
```

Comment:
How would this handle a `map`?

Reply:
It's possible to handle map, I assume of the type of the key is string or interface{} and the value of "ID" key exists.

Can entity be a map here? I assumed it can't. Does Java code allows passing dictionary to DocumentSession.Store() ?

Reply:
Yes, it does. See: https://github.com/ravendb/ravendb-jvm-client/issues/4